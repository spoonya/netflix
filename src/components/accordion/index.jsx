import React, { createContext, useState, useContext } from 'react';
import {
  Container,
  Item,
  Body,
  Header,
  Inner,
  Title
} from './styles/accordion';

const ToggleCtx = createContext();

export default function Accordion(props) {
  const { children } = props;

  return (
    <Container>
      <Inner>{children}</Inner>
    </Container>
  );
}

Accordion.Title = function FaqTitle(props) {
  const { children } = props;

  return <Title>{children}</Title>;
};

Accordion.Item = function FaqItem(props) {
  const [toggle, setToggle] = useState(false);
  const { children } = props;

  return (
    <ToggleCtx.Provider value={{ toggle, setToggle }}>
      <Item>{children}</Item>
    </ToggleCtx.Provider>
  );
};

Accordion.Header = function FaqHeader(props) {
  const { children } = props;
  const { toggle, setToggle } = useContext(ToggleCtx);

  return (
    <Header onClick={() => setToggle((prev = toggle) => !prev)}>
      {children}
      {toggle ? (
        <img src="/images/icons/close.png" alt="Close" />
      ) : (
        <img src="/images/icons/add.png" alt="Open" />
      )}
    </Header>
  );
};

Accordion.Body = function FaqBody(props) {
  const { children } = props;
  const { toggle } = useContext(ToggleCtx);

  return toggle ? <Body>{children}</Body> : null;
};
