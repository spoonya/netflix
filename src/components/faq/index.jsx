import React, { createContext, useState, useContext } from 'react';
import { Container, Item, Body, Header, Inner, Title } from './styles/faq';

const ToggleCtx = createContext();

export default function Faq(props) {
  const { children } = props;

  return (
    <Container>
      <Inner>{children}</Inner>
    </Container>
  );
}

Faq.Title = function FaqTitle(props) {
  const { children } = props;

  return <Title>{children}</Title>;
};

Faq.Item = function FaqItem(props) {
  const [toggle, setToggle] = useState(false);
  const { children } = props;

  return (
    <ToggleCtx.Provider value={{ toggle, setToggle }}>
      <Item>{children}</Item>
    </ToggleCtx.Provider>
  );
};

Faq.Header = function FaqHeader(props) {
  const { children } = props;
  const { toggle, setToggle } = useContext(ToggleCtx);

  return (
    <Header onClick={() => setToggle((prev = toggle) => !prev)}>
      {children}
      {toggle ? (
        <img src="/images/icons/close.png" alt="Close" />
      ) : (
        <img src="/images/icons/add.png" alt="Open" />
      )}
    </Header>
  );
};

Faq.Body = function FaqBody(props) {
  const { children } = props;
  const { toggle } = useContext(ToggleCtx);

  return toggle ? <Body>{children}</Body> : null;
};
