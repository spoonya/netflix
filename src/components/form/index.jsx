import React from 'react';
import {
  Container,
  Base,
  Error,
  Title,
  Text,
  TextSmall,
  Link,
  Input,
  Submit
} from './styles/form';

export default function Form(props) {
  const { children } = props;

  return <Container>{children}</Container>;
}

Form.Base = function FormBase(props) {
  const { children, onSubmit } = props;

  return <Base onSubmit={onSubmit}>{children}</Base>;
};

Form.Error = function FormError(props) {
  const { children } = props;

  return <Error>{children}</Error>;
};

Form.Title = function FormTitle(props) {
  const { children } = props;

  return <Title>{children}</Title>;
};

Form.Text = function FormText(props) {
  const { children } = props;

  return <Text>{children}</Text>;
};

Form.TextSmall = function FormTextSmall(props) {
  const { children } = props;

  return <TextSmall>{children}</TextSmall>;
};

Form.Link = function FormLink(props) {
  const { children, to } = props;

  return <Link to={to}>{children}</Link>;
};

Form.Input = function FormInput(props) {
  const { children, placeholder, type, autoComplete } = props;

  return (
    <Input placeholder={placeholder} type={type} autoComplete={autoComplete}>
      {children}
    </Input>
  );
};

Form.Submit = function FormSubmit(props) {
  const { children, disabled } = props;

  return <Submit disabled={disabled}>{children}</Submit>;
};
