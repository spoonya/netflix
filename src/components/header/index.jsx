import React from 'react';
import { Link as ReactRouterLink } from 'react-router-dom';
import {
  Container,
  Background,
  Frame,
  Logo,
  ButtonLink
} from './styles/header';

export default function Header(props) {
  const { children, bg } = props;

  return bg ? children : <Background>{children}</Background>;
}

Header.Frame = function HeaderFrame(props) {
  const { children } = props;

  return <Container>{children}</Container>;
};

Header.Logo = function HeaderLogo(props) {
  const { to, src, alt } = props;

  return (
    <ReactRouterLink to={to}>
      <Logo src={src} alt={alt} />
    </ReactRouterLink>
  );
};

Header.ButtonLink = function HeaderButtonLink(props) {
  const { children, to } = props;

  return <ButtonLink to={to}>{children}</ButtonLink>;
};
