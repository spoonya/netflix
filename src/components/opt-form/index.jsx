import React from 'react';
import { Container, Input, Button, Text } from './styles/opt-form';

export default function OptForm(props) {
  const { children } = props;

  return <Container>{children}</Container>;
}

OptForm.Input = function OptFormInput() {
  return <Input />;
};

OptForm.Button = function OptFormButton(props) {
  const { children } = props;

  return (
    <Button>
      {children}
      <img src="/images/icons/chevron-right.png" alt="Try now" />
    </Button>
  );
};

OptForm.Text = function OptFormText(props) {
  const { children } = props;

  return <Text>{children}</Text>;
};
