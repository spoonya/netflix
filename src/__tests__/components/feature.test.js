import React from 'react';
import { render } from '@testing-library/react';
import { Feature } from '../../components';

describe('<Feature />', () => {
  it('renders the <Feature />', () => {
    const { container, getByText } = render(
      <Feature>
        <Feature.Title>Unlimited movies, TV shows and more.</Feature.Title>
        <Feature.SubTitle>Watch anywhere. Cancel anytime.</Feature.SubTitle>
        <Feature.Disclaimer>
          Only new members are eligible for this offer.
        </Feature.Disclaimer>
      </Feature>,
    );

    expect(getByText('Unlimited movies, TV shows and more.')).toBeTruthy();
    expect(getByText('Watch anywhere. Cancel anytime.')).toBeTruthy();
    expect(
      getByText('Only new members are eligible for this offer.'),
    ).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
