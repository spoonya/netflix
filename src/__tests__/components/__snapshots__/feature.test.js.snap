// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feature /> renders the <Feature /> 1`] = `
<div
  class="feature__Container-sc-6swrp6-0 gazZKD"
>
  <h1
    class="feature__Title-sc-6swrp6-1 gGONpd"
  >
    Unlimited movies, TV shows and more.
  </h1>
  <h2
    class="feature__SubTitle-sc-6swrp6-2 eduQxj"
  >
    Watch anywhere. Cancel anytime.
  </h2>
  <p
    class="feature__Disclaimer-sc-6swrp6-3 vpAGK"
  >
    Only new members are eligible for this offer.
  </p>
</div>
`;
